frontend/src/metabase-types/api/settings.ts

```diff
@@ -359,6 +359,35 @@ export type UserSettings = {
   "show-updated-permission-banner": boolean;
 };
 
+/**
+ * Important distinction between `null` and `undefined` settings values.
+ *  - `null` means that the setting actually has a value of `null`.
+ *  - `undefined` means that the setting is not available in a certain context.
+ *
+ * Further longer explanation:
+ *
+ * `null` is a backend way of deleting a value, which can backfire on frontend if we are not
```

Comment:
can you clarify "deleting" in this context?
Deleting the setting (as in we don't need it anymore so we remove the defsetting) or "un-setting" it (as in the user removed clicked on "reset to default" or similar)?

Reply:
I am confused by "deleting a value", in js deleting means removing it from the object, which means it won't show up in the object (and technically the value is now undefined):
<img width="1023" alt="image" src="https://github.com/user-attachments/assets/4395a5e3-63b9-4b7b-a0fa-546c4f167f82">

Do you have an example of when the BE "deletes" a value and in which way it can backfire?

Reply:
I'll clarify this in the comment.

It helps if you look at types.
`"last-used-native-database-id"?: number | null;`

`?` - this is user level setting, it might not be available for unauthenticated users
`null` - default value, or "no value" (this is where the confusion might kick in because on FE we're used to think of this as `undefined` more often than not
`number` - the value has been set

If you want to delete/reset it, you'd change it to `nulll`.

Makes more sense now?

Reply:
Does this make it less confusing?
https://github.com/metabase/metabase/pull/39223/commits/cadf6e3098372073dbe48eee41e68d7ff58022b0